crates/cairo-coverage/src/input/mappings/statement_map.rs

```diff
@@ -0,0 +1,70 @@
+use crate::input::data_loader::deserialize;
+use crate::input::data_loader::types::{
+    CodeLocation, CoverageAnnotations, LineRange, ProfilerAnnotations,
+};
+use crate::types::{FileLocation, FunctionName};
+use anyhow::Result;
+use cairo_lang_sierra::debug_info::Annotations;
+use cairo_lang_sierra::program::StatementIdx;
+use derived_deref::Deref;
+use std::collections::HashMap;
+
+const PROFILER_NAMESPACE: &str = "github.com/software-mansion/cairo-profiler";
+const COVERAGE_NAMESPACE: &str = "github.com/software-mansion/cairo-coverage";
+
+#[derive(Deref)]
+pub struct StatementMap(HashMap<StatementIdx, StatementOrigin>);
+
+#[derive(Clone, Eq, PartialEq)]
+pub struct StatementOrigin {
+    pub function_name: FunctionName,
+    pub file_location: FileLocation,
+    pub line_range: LineRange,
+}
+
+impl TryFrom<&Annotations> for StatementMap {
+    type Error = anyhow::Error;
+
+    fn try_from(annotations: &Annotations) -> Result<Self, Self::Error> {
+        let CoverageAnnotations {
+            statements_code_locations,
+        } = deserialize::by_namespace(annotations, COVERAGE_NAMESPACE)?;
+
+        let ProfilerAnnotations {
+            statements_functions,
+        } = deserialize::by_namespace(annotations, PROFILER_NAMESPACE)?;
+
+        Ok(Self(
+            statements_code_locations
+                .into_iter()
+                .filter_map(|(key, code_locations)| {
+                    statements_functions
+                        .get(&key)
+                        .and_then(|function_names| {
+                            find_statement_origin(&code_locations, function_names)
+                        })
+                        .map(|statement_origin| (key, statement_origin))
+                })
+                .collect(),
+        ))
+    }
+}
+
+fn find_statement_origin(
+    code_locations: &[CodeLocation],
+    function_names: &[FunctionName],
+) -> Option<StatementOrigin> {
+    code_locations
+        .iter()
+        .zip(function_names)
+        // TODO: We should probably filter by path to user project not by path to cache
```

Comment:
Will be done in separate PR

Reply:
Is there an issue?

Reply:
https://github.com/software-mansion/cairo-coverage/issues/26